overlord/snapstate/handlers_link_test.go

```diff
@@ -1229,6 +1230,88 @@ func (s *linkSnapSuite) TestDoLinkSnapdDiscardsNsOnDowngrade(c *C) {
 	c.Check(lp.instanceNames, DeepEquals, []string{"snapd"})
 }
 
+func (s *linkSnapSuite) TestDoLinkSnapdRemovesAppArmorProfilesOnSnapdDowngrade(c *C) {
```

Comment:
let me know if you want me to debug more locally

Reply:
this passes for me locally and in the spread tests here so perhaps something has changed on master - also since we mock the backend it won't actually do anything as part of the operation so I don't think we can test removal of system-key

Reply:
dug a bit, it's because I hadn't run mkversion.sh in a while, as I tend to run test directly with go test. The test would be slightly more robust if it mocked snapdtool.Version as well with snapdtool.MocklVersion

Reply:
ah good point - added in 7301a9f2a6